### ISSUE TITLE
403 on map.init when running with capacitor 3.x on iOS

### ISSUE THREAD

**Issue by AVatch** [mapbox/mapbox-gl-js#11170]
Hey there - so I wanted to experiment with Mapbox and ran it via [Capacitor](https://capacitorjs.com) which would serve the app on iOS/android via web view.

However, I seem to be hitting an issue with the token permission checks. On Chrome I can init the map and it works was expected but on iOS devices I get the following error in the logs

```
 [error] - {"status":403,"url":"https://api.mapbox.com/v4/mapbox.mapbox-streets-v8,mapbox.mapbox-terrain-v2.json?secure&access_token=pk.MY-TOKEN"}
```

When I put this url in the browser, I also get an error, so I try to use a token that has ```Download:Read``` scope and that results in a successful 200. However, if I use a secret token in the browser OR capacitor, I get an exception thrown to not use a secret token.

I've made sure to allow the following urls in my token
```capacitor://localhost```
```localhost```
```api.mapbox.com```

I would expect the behavior to be consistent with Safari behavior.

I've attached the network request as well if it is helpful

<img width="1440" alt="Screen Shot 2021-10-22 at 4 48 21 PM" src="https://user-images.githubusercontent.com/1807958/138538073-ac90eb95-b3fa-4180-8206-efe8ed51bb1c.png">

**mapbox-gl-js version**: ^2.5.1

### Steps to Trigger Behavior

 1. ```ionic build```
 2. ```ionic capacitor sync```
 3. ```ionic capacitor open ios```
 4. Run in simulator
 5. Connect Safari via web dev tools
 6. Observe results when clicking button to load map

### Link to Demonstration

https://github.com/AVatch/bug-mapbox-capacitor-403

### Expected Behavior

The map should init as it does on browsers

### Actual Behavior

The map does not init and throws a 403

Thanks for checking this out! :D

**Comment by gazpachu** [comment]
I have the same issue but with a Cordova app

**Comment by avpeery** [comment]
Hi @AVatch - please see the documentation on [how to use mapbox securely](https://docs.mapbox.com/help/troubleshooting/how-to-use-mapbox-securely/#access-tokens), [scopes]( ahttps://docs.mapbox.com/accounts/guides/tokens/#scopes) and [URL restrictions](https://docs.mapbox.com/accounts/guides/tokens/#url-restrictions). If these sources do not help, then it could be a with `capacitor://localhost` not being considered a secure source from Apple (see closed issue [#9680](https://github.com/mapbox/mapbox-gl-js/issues/9680)). Thanks!
